code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -435,4 +521,16 @@ ExecutorService getExecutor() {
             return executorService;
         }
     }
+
+    /**
+     * Gets application info from the {@link AndroidSystemInfoService} and builds an application identifier string.
+     *
+     */
+    private void getApplicationIdentifier() {
+        final String applicationName =  systemInfoService.getApplicationName();
+        final String applicationVersion = systemInfoService.getApplicationVersion();
+        final String applicationBuildNumber = systemInfoService.getApplicationVersionCode();
+        final StringBuilder applicationIdentifierStringBuilder = new StringBuilder().append(applicationName).append(applicationVersion).append(applicationBuildNumber);
+        this.applicationIdentifier = applicationIdentifierStringBuilder.toString().replaceAll("  ", " ").replaceAll("()", "").trim();
```

Comment:
Minor - Can we make this utility method and move the line to constructor
this.applicationIdentifier = getApplicationIdentifier()